src/main/kotlin/com/gamenews/models/Article.kt

```diff
@@ -1,31 +1,24 @@
 package com.gamenews.models
 
-import java.util.concurrent.atomic.AtomicInteger
+import org.bson.codecs.pojo.annotations.BsonId
+import org.bson.types.ObjectId
 
 /**
  * Model class for the news articles
  */
-class Article
-private constructor(
-    val id: Int,
+data class Article(
+    @BsonId
+    val id: String,
     var title: String,
     var body: String
 ) {
     companion object {
         // automatically generate unique ID
-        private val idCounter = AtomicInteger()
+        private val idCounter = ObjectId().toString()
 
         /**
          * Create a new article with an id, title, and body
          */
-        fun newEntry(title: String, body: String) = Article(idCounter.getAndIncrement(), title, body)
+        fun newEntry(title: String, body: String) = Article(idCounter, title, body)
```

Comment:
I have the suspicion that `idCounter` will have the same value for every entity (insert thinking face). Could we verify that the `idCounter` changes for every new entry that is created.

Reply:
Done